File: javascript/insertionSort.js
let array = [];
let dim = Math.floor(Math.random() * 1000);
for (let i = 0; i < dim; i++) {
	array[i] = Math.floor(Math.random() * 100000);
}
let n = array.length;
console.log("Unsorted Array: " + array);

function insertionSort(A, n) {
	for (let i = 1; i < n; i++) {
		let key = A[i];
		let j = i - 1;
		while (j >= 0 && A[j] > key) {
			A[j + 1] = A[j];
			j = j - 1;
		}
		A[j + 1] = key;
	}
}

var startTime = performance.now();
insertionSort(array, n);
var endTime = performance.now();

console.log("Sorted Array: " + array);
console.log("\nExecution Time: " + (endTime - startTime) + " ms!");

File: python/insertionSort.py
import time
import random 

def insertionSort(A):
	for i in range(1, len(A)):
		key = A[i]
		j = i-1
		while (j >= 0 and A[j] > key):
			A[j+1] = A[j]
			j = j-1
		A[j+1] = key

A = [random.randint(1, 100) for _ in range(10)]

print("Unsorted array:", A)

startTime = time.time()

insertionSort(A)

endTime = time.time()
executionTime = endTime - startTime

print("Sorted array:", A)
print(f"Execution time: {executionTime:.6f} seconds")
       